### ISSUE TITLE
In DIY mode your tool.exe doesn't find the devices

### ISSUE THREAD

**Issue by alexbk66** [itead/Sonoff_Devices_DIY_Tools#8]
Here's what I do:
1. Connect to devices NOT in DIY mode (no jumper) to verify firmware version, all good:
![image](https://user-images.githubusercontent.com/2533315/59899497-49474e00-9438-11e9-9e53-5819be5c828b.png)
![image](https://user-images.githubusercontent.com/2533315/59899507-5401e300-9438-11e9-985c-cb321d73dd15.png)
![image](https://user-images.githubusercontent.com/2533315/59899514-582e0080-9438-11e9-95a9-43203d51f848.png)
2. Put the jumpers back on, enable hotspot, connect my laptop to the hotspot, both devices connect too:
![image](https://user-images.githubusercontent.com/2533315/59899526-67ad4980-9438-11e9-948e-2036c0136c2c.png)
3. Use your tool_01DIY85(v1.0.0).exe - nothing:
![image](https://user-images.githubusercontent.com/2533315/59899616-a8a55e00-9438-11e9-87bd-4f315bdd00b4.png)
Very disappointing and frustrating.  I spent now many days trying to make this work. 
Please do something.  May be can you send me 4 new devices with installed and tested firmware.

**Comment by ZZLinvec** [comment]
Currently, the device supports 2.4ghz, and I see that it is 5GHz in your screenshot. I think I can find it if I change it here.

**Comment by alexbk66** [comment]
No, it's sharing from another network called "MyRepublic 5GHz" (it's actually Ethernet connection, not WiFI) - but the hotspot if 2.4GHz.

So please do something, it's been too long

________________________________

**Comment by ZZLinvec** [comment]
Can I understand that this is a 2.4GHz hotspot

**Comment by alexbk66** [comment]
Yes

________________________________

**Comment by alexbk66** [comment]
You can see on the same screenshot that 3 devices are connected - and I said so in my email: "onnect my laptop to the hotspot, both devices connect too"
[cid:6ab3b770-9360-493e-bfc0-2f3edbe5a952]

________________________________

**Comment by ZZLinvec** [comment]
I'll make a log version, you run it, I'll look at the log, ok?I think this is my software what is wrong.

**Comment by alexbk66** [comment]
Ok, thank you

________________________________

**Comment by ZZLinvec** [comment]
hi,I have uploaded the log version to /tool

**Comment by alexbk66** [comment]
So what do I need to do now?
________________________________

**Comment by ZZLinvec** [comment]
You run it and see if you can find the device

**Comment by alexbk66** [comment]
Not much in the log :)
![viber_image_2019-06-24_21-25-43](https://user-images.githubusercontent.com/2533315/60015760-9a20a600-96c7-11e9-8cf0-82be185360d6.jpg)
![2019-06-24](https://user-images.githubusercontent.com/2533315/60015767-9e4cc380-96c7-11e9-8c50-cd2ef9c815d4.png)

**Comment by ZZLinvec** [comment]
Is it possible to set your router to only 2.4GHz?I want to make sure that both the computer and the device are running at 2.4GHz.

**Comment by alexbk66** [comment]
But when devices and my computer connect to hotspot - router is not involved at all?

________________________________

**Comment by ZZLinvec** [comment]
Sorry, by that I mean setting the hotspot to 2.4GHZ

**Comment by alexbk66** [comment]
Hotspot must be 2.4Ghz - the devices are connected to it!  It's definitely some issue in the firmware.  
Adding logging in you "tool" app doesn't help - I know your devices do not respond to mDNS requests.

**Comment by alexbk66** [comment]
Any more ideas, except for WiFi frequency?

**Comment by chatainsim** [comment]
Same issue here. the app "tool" doesn't find any sonoff devices.

**Comment by anton-yakunin** [comment]
Same problem with three new Basic R3 controllers. Looks like a big in firmware. Please see tickets ##324553 and 323823 filed with Itead support.
1. The device does not connect to sonoffDiy hotspot (no matter where - phone or PC), and yes, I know it has to be 2.4Ghz.
2. Upgrading to the latest available firmware version (3.1.0, if I remember correctly) kills the DIY mode for good - with the jumper on it does not get into the configuration mode anymore. The eWelink continues to work fine though.

**Comment by ZZLinvec** [comment]
HI,If the device is not connected to WIFI and in DIY mode, it will be a single flash.If the route is connected it will be a double flash.

- single flash（   _ - _ _ _ _ _ _   ）

- double flash(  _-_-_ _ _ _ _ _   )

**Comment by ZZLinvec** [comment]
Do you have two network CARDS?If so, you need to set the network card connected to "sonoffDiy" as the preferred one.

**Comment by ZZLinvec** [comment]
At present, DIY only supports these devices.
Basic R3, RFR3 and Mini

**Comment by alexbk66** [comment]
> Do you have two network CARDS?If so, you need to set the network card connected to "sonoffDiy" as the preferred one.

Most laptops have two network cards - WiFi and Ethernet.  So, where exactly do I "set the network card connected to "sonoffDiy" as the preferred one."?

**Comment by alexbk66** [comment]
I created a HomeSeer plugin for Shelly devices, they are very simple to configure and add to WiFi network.
@ZZLinvec - you should probably have a look at their implementation and do something similar. 
https://shop.homeseer.com/products/alexbk66-ak-shelly-software-plug-in-for-hs3

**Comment by alexbk66** [comment]
Finally I managed to discover my devices!  After flashing FW 3.3 and disabling Ethernet adaptor on my laptop.

**Comment by chatainsim** [comment]
Same for me.
Thanks

**Comment by rradar** [comment]
It's so f*cking not funny :disappointed:

Did itead research since june 2017 (wifi password for sonoffDiy) how to make this flashing the most complicated thing in the world that people will still use their cl0ud sh!t?

**Comment by dastrix80** [comment]
This plain and simple does not work. Ive tried windows 10, same settings, all correct. laptop connects, sonoff mini will not.

**Comment by frank3791** [comment]
Same issue here. Using mini, de 3.3, jumper set,  two gaat led flash mode,  2.4GHz only router,  pc wifi metric set to 1, pc lan disabled, esp device found by angy ip scanner, ping result is ok, dit tool finds no device. Log window of diy tool only shows 4 strange characters (equal as reported above in screenshot). Any suggestions?

**Comment by chatainsim** [comment]
How did you connect pc and Sonoff ?
I've used the Tethering off my phone to connect both the sonoff and my pc on the same wifi.

**Comment by frank3791** [comment]
Its working now after many trial and error. On PC, I switched my dual band wifi adapter to (4) 2.4GHz 802.11b/g. Then the adaper restarts and it cannot connect. Then I switch back to (8) dual band 802.11a/b/g and adapter restarts and it can connect to sonofDIY SSID. Also the tool shows the mini device and I can control it.
I hope it this can help someone.

**Comment by GaryMilne** [comment]
Trying to Flash a new Sonoff Mini. Have got as far as all devices connected to a 2.4GHz hotspot. Can see that the Sonoff gets an IP address and I can ping that address but the DIY mode tool never discovers.  Have tried the various solutions listed here such as disabling the ethernet adapter and running the (log) version but the software fails to discover the device.

I don't understand the point of DIY mode if A) It is as cumbersome as it is and B) If it is not being supported. This problem was reported months ago and there seems to be no progress.  Why not add the ability to manually add the device by IP address.  Anyone that is taking on this process is more than capable to downloading an IP scanner and finding the Sonoff.

**Comment by steveizzle** [comment]
Fyiy, I had the same problem with Windows 7 and could fix it by allowing the exe at Windows firewall to connect to private and public networks (had a dialogue at application start)...

**Comment by Snoopy1368** [comment]
I have the same problems DIY to a Mini the device is visible on the phone hot spot so it is the correct network frequency it is possible to view the device IP address and Mac Address it is possible to log another device onto the hotspot and see the device.  I also ran ewelink over the hot spot and it found the device and I could turn it on and off from there but I cant find it in DIY mode Tool.   Looks like I have to live dangerously and go to a hardware flash which is a "bugger" on the mini because there is no provision for pin connectors.   Safe bet they don't want you using the DIY Tasmota solution or they would have allowed it as a simple software upgrade.  With the current way the CCP is behaving how long before they decide to shut down eWelink.  Time to look for an alternative.

**Comment by mikila85** [comment]
Confirmed, resolved:

![image](https://user-images.githubusercontent.com/3965957/105911013-5f64b980-6032-11eb-8701-7b8cc5990752.png)
![image](https://user-images.githubusercontent.com/3965957/105911025-625faa00-6032-11eb-92d0-3025d2dbc703.png)
![image](https://user-images.githubusercontent.com/3965957/105911042-668bc780-6032-11eb-97fb-c902a2e7e29e.png)

**Comment by Serious667** [comment]
Hi
FYI, I've got the same problem - helped restart of the network card (disable/enable) but after connecting to different network and going back to the AP, I had to apply restart again.
